src/Voron/Data/DirectAddScope.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Voron.Data.BTrees;
+
+namespace Voron.Data
+{
+    public unsafe class DirectAddScope : IDisposable
+    {
```

Comment:
This being a  class is too costly. It should be a struct instead (you are not supposed to keep these scopes around anyways).